File: python/statgen/_verbosity.py
import sys

_LEVELS = ("quiet", "info")
_VERBOSITY = "info"


def set_verbosity(level: str) -> None:
    global _VERBOSITY
    if level not in _LEVELS:
        raise ValueError("verbosity must be one of: quiet, info")
    _VERBOSITY = level


def get_verbosity() -> str:
    return _VERBOSITY


def info(message: str) -> None:
    if _VERBOSITY == "info":
        print(message, file=sys.stderr)

File: R-package/R/verbosity.R
.statgen_state <- new.env(parent = emptyenv())
.statgen_state$verbosity <- "info"

set_verbosity <- function(level) {
  if (!is.character(level) || length(level) != 1L || is.na(level)) {
    stop("level must be a character scalar", call. = FALSE)
  }
  if (!(level %in% c("quiet", "info"))) {
    stop("level must be one of 'quiet' or 'info'", call. = FALSE)
  }
  .statgen_state$verbosity <- level
  invisible(level)
}

get_verbosity <- function() {
  .statgen_state$verbosity
}
